src/org/labkey/test/components/ui/EntityBulkInsertDialog.java

```diff
@@ -136,6 +176,19 @@ protected ElementCache elementCache()
 
     protected class ElementCache extends ModalDialog.ElementCache
     {
+        public WebElement formRow(String fieldKey)
+        {
+            return Locator.tagWithClass("div", "row")
+                    .withChild(Locator.tagWithAttribute("label", "for", fieldKey))
+                    .findElement(this);
+        }
+
+        public Input textInput(String fieldKey)
+        {
+            WebElement inputEl = textInputLoc.findElement(elementCache().formRow(fieldKey));
+            return new Input(inputEl, getDriver());
+        }
```

Comment:
Why not replace the locator in the getTextField and setTextField with this locator.

Reply:
good call

Reply:
TBH I'd avoided doing that because doing it would require me to crack open SM tests.
It turns out that getTextField as implemented took the column label text in the dialog, not the fieldKey/column name.
Buuuuut, I'll do it

Reply:
Thanks!